tests/unit/test_client.py

```diff
@@ -0,0 +1,102 @@
+import os.path
+
+from botocore.exceptions import ClientError
+import pytest
+
+from moto import mock_s3
+from dotenv import load_dotenv
+
+import re
+from IIIFingest.client import Client
+
+load_dotenv()
+
+
+@mock_s3
+class TestClient:
+    test_aws_profile = os.getenv('TEST_AWS_PROFILE', default="tester")
+    # must match client params e.g. account space and environment - see conftest.py
+    bucket_name = os.getenv('TEST_BUCKET', default="edu.harvard.huit.lts.mps.test-testing-space-dev")
+    
+    # Define manifest-level metadata
+    manifest_level_metadata = {
+        "labels": ["Test Manifest"],
+        "summary": "A test manifest for ingest",
+        "rights": "http://creativecommons.org/licenses/by-sa/3.0/",
+    }
+
+    def test_client_upload(self, test_images, boto_session, test_client):
+        boto_session.resource('s3').create_bucket(Bucket=self.bucket_name)
+
+        image_path = test_images["27.586.1-cm-2016-02-09.tif"]["filepath"]
+
+        client = test_client
+
+        images = [{"label": "Test Image", "filepath": image_path}]
+        assets = client.upload(images, s3_path="testing")
+
+        assert assets is not None
+        assert assets[0].filepath == image_path
+        
+    def test_client_fail_upload(self, test_images, boto_session, test_client):
+        with pytest.raises(ClientError):
+            boto_session.client('s3').delete_bucket(Bucket=self.bucket_name)
+            image_path = test_images["27.586.1-cm-2016-02-09.tif"]["filepath"]
+
+            client = test_client
+
+            images = [{"label": "Test Image", "filepath": image_path}]
+            assert client.upload(images, s3_path="testing")
+
+    def test_client_create_manifest(self, test_images, boto_session, test_client):
+        boto_session.resource('s3').create_bucket(Bucket=self.bucket_name)
+        client = test_client
+        image_path = test_images["27.586.1-cm-2016-02-09.tif"]["filepath"]
+        images = [{"label": "Test Image", "filepath": image_path}]
+        assets = client.upload(images, s3_path="testing")
+        """
+        Example of the assets list of dict
+        assets = [
+            {
+                'asset_id': 'testTMf2nGLoepAPr2tvEpw6EA',
+                'extension': '.tiff',
+                'filepath': '/Users/vtan/harvard-atg/lts-iiif-ingest-service/images/27.586.1-cm-2016-02-09.tif',
+                'format': 'image/tiff',
+            }
+        ]
+        """
+
+        manifest = client.create_manifest(
+            manifest_level_metadata=self.manifest_level_metadata, assets=assets
+        )
+        root_id = "https://nrs-dev.lib.harvard.edu/URN-3:TEST:"
```

Comment:
I'm not quite clear on where this value comes from. Is this the same as the `MPS_MANIFEST_BASE_URL` from the settings (formatted for the dev environment)?

Reply:
Yes, it is the same one as `MPS_MANIFEST_BASE_URL`. I hard coded it for now since I couldn't understand how to pass parameters to the `test_client` fixture. [Link to docs](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#fixture-parametrize)

Reply:
What about using the base URLs configured by the client? The client has an `asset_base_url` and `manifest_base_url`, and I'm not sure which one you'd need here, but if you assume the client is being constructed correctly (that should be a separate test to verify), then you could rely on those values for this particular test. That would make it a little more self-documenting as far as what's expected for the `manifest["id"]` and where it's coming from. Alternatively, you could import the `MPS_MANIFEST_BASE_URL` from `IIIFIngest.settings` and just format it manually like the client is doing.

Reply:
Yes, that makes much more sense! I was looking to test the manually formatted `manifest_url` against the ones generated from`create_manifest` to see if they are correctly generated. So I think manually generating the manifest from `MPS_MANIFEST_BASE_URL` would be ideal for this particular test.